src/main/java/frc/robot/posev2/RobotPose.java

```diff
@@ -0,0 +1,53 @@
+package frc.robot.posev2;
+
+import edu.wpi.first.wpilibj.geometry.Pose2d;
+import edu.wpi.first.wpilibj.geometry.Rotation2d;
+
+
+public class RobotPose {
+    private RobotPosition robotPosition;
+    private VisionData visionData;
+    private Pose2d robotPose2d;
+    private WheelColorValue wheelColor;
+    private TargetLocation targetLocation;
+    
+    public RobotPose(double f, double h, double t, VisionData vData){
+        robotPosition = new RobotPosition(f, h, t);
+        visionData = vData;
+        robotPose2d = new Pose2d(f, h, new Rotation2d(t));
+        wheelColor = null;
+        targetLocation = new TargetLocation(vData);
```

Comment:
i like keeping the code to calculate the target location separate from the vision data. however, i think you should have a separate class that uses a simple function to do it. IE

TargetLocation = new VisionDataProcessor().compute(VisionData)